### PR TITLE
Rubocop and Travis CI updates. Remove the Danger gem.

### PR DESCRIPTION
**Pull request by jonatack**
This gem looks like it needs a little update love :heart: but first we need to get past all the new Rubocop rules in order to run the test suite, and we need the CI build to work:

- Make minimum changes to have the cop pass without warnings.
- Update the change log.
- Update the Ruby version for Travis so the CI still runs.
- Remove the out-of-date Danger gem which was blocking the CI build. It hasn't seen maintenance in a couple of years and the PR activity here is low enough that it doesn't appear to be acutely needed.

Cheers.

### CODE REVIEW

**Comment by jonatack** [comment]
The build for PR #70 should be green once this is merged.

**Comment by dblock** [comment]
For rubocop I generally just `rubocop -a ; rubocop --auto-gen-config` and call it a day. I would avoid `#rubocop disable` unless we absolutely think it's required.

File: .travis.yml [hunk 1]
```diff
@@ -10,17 +10,14 @@ before_install:
...
+  - 2.4.1
 
 gemfile:
   - gemfiles/mongoid-3.1.gemfile
   - gemfiles/mongoid-4.0.gemfile
   - gemfiles/mongoid-5.0.gemfile
   - gemfiles/mongoid-6.0.gemfile
 
-before_script:
-  - bundle exec danger
```
[the remark this answers is not in the document]
Comment (jonatack) [reply .travis.yml]: Danger appeared to be breaking the CI build. I'll try adding it back in.

**Comment by jonatack** [comment]
Agreed, and that worked for me locally, but it was not enough for the CI build. Adding the disable comment solved it. I can try again without the comment to see.

**Comment by jonatack** [comment]
Hi @dblock, added the Danger gem back in and once again it appears to break the CI. Please advise?

**Comment by dblock** [comment]
The error is posted above. It says that one of the lines in changelog is invalid. It's nitpicking, it doesn't like the period in `Remove the Danger gem. - [@jonatack](...)`. You don't need that line anyway, no need to mention that CI works again :) We definitely do appreciate your work though!

**Comment by jonatack** [comment]
@dblock when the CI was broken by Danger, I naturally looked at the build log and saw:

```
$ bundle exec danger
Danger has failed this build. 
Found 1 error.
The command "bundle exec danger" failed and exited with 1 during .
Your build has been stopped.
```
There was no mention that it could be due to the pollution that Danger is adding here.
I've seen Danger respond with 3 different messages to my commits.
In each case, the messages were unhelpful and confusing, when not downright wrong.
This gem is causing more lost developer time than it is helping.
I'm abandoning this PR and moving this gem back to passing builds and will just use my fork.
Cheers.

**Comment by dblock** [comment]
cc: @orta ^ danger should probably say why it's failing on the command line too

@jonatack if you want to improve messages in danger-changelog, it's [here](https://github.com/dblock/danger-changelog)

Thanks for your help @jonatack.

**Comment by orta** [comment]
Pretty sure if you run with `--verbose` it will, can't remember how the ruby version works anymore

**Pull request closed without merging**